src/memray/_memray/hooks.h

```diff
@@ -4,16 +4,21 @@
 #define PY_SSIZE_T_CLEAN
 #include <Python.h>
 
+#include <dlfcn.h>
+#include <sys/mman.h>
 #include <sys/types.h>
 
 #include <cstdlib>
 #include <iostream>
-#include <malloc.h>
 
-#include "elf_utils.h"
-#include <dlfcn.h>
-#include <sys/mman.h>
-#include <sys/prctl.h>
+#ifdef __linux__
```

Comment:
Hm - can we encapsulate the Linux version all into a separate header that's conditionally included, rather than littering `#ifdef __linux` all over this file (can be saved for a later cleanup PR, if you prefer)

Reply:
Let's do this in a cleanup PR if you don't mind, just to avoid having to restructure a lot and go into merge conflicts :S